### ISSUE TITLE
Option name does not change when edited

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2582]
### Expected behavior
When you change the option title (not the label) it should change elsewhere

### Actual Behavior
It does not change unless you delete the existing contents and then click away and then replace it.
![basic_t-shirt](https://user-images.githubusercontent.com/72819/28450308-2595a6ec-6e19-11e7-8d14-deec6987f41a.png)

### Steps to Reproduce the Behavior
1. Login as admin
1. Create a new product
1. Set the Variant label
1. Change title, label, quantity and price.
1. Observe that the option title shown does not change but remains "Untitled Option" even when you publish

### Versions
```
Node: 7.10.0
NPM: 5.0.3
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.4.0
Reaction branch: development
Docker: 17.06.0-ce
```

**Comment by aaronjudd** [comment]
This is also displaying an uncaptured error on the server console, which should be trapped.

**Comment by zenweasel** [comment]
Closed via #2687
